File: 삼성 코테 기출/7568 덩치.c
#include <stdio.h>

int arr[100][2];
int n;
int ans[100] = { 0 };

void func(void)
{
	for (int i = 0; i < n; i++) {
		int cnt = 1;
		for (int j = 0; j < n; j++) {
			if ((arr[i][0] < arr[j][0]) && (arr[i][1] < arr[j][1])) cnt += 1;
		}
		ans[i] = cnt;
	}
}

int main(void)
{
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		scanf("%d %d", &arr[i][0], &arr[i][1]);
	}
	func();
	for (int i = 0; i < n; i++) {
		printf("%d ", ans[i]);
	}
	return 0;
}

File: 삼성 코테 기출/7568 덩치.py
n = int(input())
arr = []
for i in range(n) :
  arr.append(list(map(int, input().split())))

ans = []

for a in arr :
  cnt = 0
  for b in arr :
    if a[0] < b[0] and a[1] < b[1] :
      cnt += 1
  cnt += 1
  ans.append(cnt)

print(*ans)